feat: add CLI workflow run subcommand - resolves #2220

alina-dvloper opened the pull request:
# Add CLI workflow run subcommand

## Resolves
Closes #2220

## Summary
 This PR implements the missing `skyvern run workflow` CLI subcommand that enables programmatic workflow execution from the command line, bringing feature parity with the Web UI and REST API.

## What was implemented
-  `skyvern run workflow` command with all required parameters
-  Support for both local YAML files (`--file`) and stored workflows (`--workflow-id`)
-  Multiple parameter formats (`--param key=value`, `--data JSON`)
-  Automation features (`--watch`, `--output`) for CI/CD integration
-  Comprehensive error handling and validation
-  Direct HTTP API integration for reliable execution

## Usage Example
```bash
skyvern run workflow -f test_workflow.yaml --param env=prod --param user=john

```

## Testing
-  CLI help and validation work correctly
-  Parameter parsing handles all formats
-  File loading and YAML validation functional
-  HTTP API integration makes correct calls
-  Error handling provides helpful feedback
<!-- ELLIPSIS_HIDDEN -->

----

> [!IMPORTANT]
> Adds `skyvern run workflow` CLI subcommand for executing workflows with support for local YAML files and stored workflows, including comprehensive error handling and HTTP API integration.
> 
>   - **New Feature**:
>     - Adds `skyvern run workflow` CLI subcommand in `skyvern/cli/run_commands.py` for executing workflows.
>     - Supports local YAML files (`--file`) and stored workflows (`--workflow-id`).
>     - Allows multiple parameter formats (`--param key=value`, `--data JSON`).
>     - Includes automation options (`--watch`, `--output`) for CI/CD.
>     - Integrates with HTTP API for execution.
>   - **Error Handling**:
>     - Validates input parameters and handles errors with descriptive messages.
>     - Handles HTTP response errors, including 401, 403, and 404 status codes.
>   - **Testing**:
>     - Adds `test_workflow.yaml` for testing the new command.
>     - Ensures CLI help, parameter parsing, and file loading are functional.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=Skyvern-AI%2Fskyvern&utm_source=github&utm_medium=referral)<sup> for 3a12c41728b51f18dff7b659e883e34ffd8883da. You can [customize](https://app.ellipsis.dev/Skyvern-AI/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

<!-- ELLIPSIS_HIDDEN -->

alina-dvloper committed:
feat: add CLI workflow run subcommand - resolves #2220

alina-dvloper committed:
fix: address code review

deveshruttala-edu commented:
👋 Hey there! This looks like something Hyperbrowser could assist with. We're building an AI-native browser that automates workflows, integrates smart tools, and enhances productivity directly inside the browser. If you're tackling tasks like this often, Hyperbrowser might be just what you need.

🚀 Learn more and try it out: https://www.hyperbrowser.ai/
We’d love your feedback!

deveshruttala-edu commented:
👋 Hey there! This looks like something Hyperbrowser could assist with. We're building an AI-native browser that automates workflows, integrates smart tools, and enhances productivity directly inside the browser. If you're tackling tasks like this often, Hyperbrowser might be just what you need.

🚀 Learn more and try it out: https://www.hyperbrowser.ai/
We’d love your feedback!

alina-dvloper committed:
fix: add missing proxy_location to API payload and fix test workflow ID

- Add proxy_location parameter to CLI workflow command
- Add proxy_location to API request payload (resolves ellipsis-dev issue)
- Fix empty workflow_id in test_workflow.yaml
- Remove temporary test files

alina-dvloper committed:
Merge branch 'main' into feature/add-workflow-cli-command

alina-dvloper commented:
@Skyvern-AI team - All ellipsis-dev issues have been resolved! 

Changes made:

-  Added missing proxy_location parameter and API payload integration

-  Fixed empty workflow_id in test_workflow.yaml  

-  All code quality checks passing (ruff, mypy, isort)

-  Core functionality tested and verified

This PR fully implements the missing skyvern run workflow CLI subcommand requested in issue #2220. Ready for review!

suchintan committed:
Merge branch 'main' into feature/add-workflow-cli-command
